src/util/records.py

```diff
@@ -620,7 +620,10 @@ def if_unchanged(cr, xmlid, callback, interval="1 minute", **kwargs):
     :param str interval: interval after `create_date` on which a record is considered as
                         _changed_, see :func:`~odoo.upgrade.util.misc.is_changed`
     """
-    if not is_changed(cr, xmlid, interval=interval):
+    changed = is_changed(cr, xmlid, interval=interval)
+    if changed is None:
+        return  # do not create a record or xmlid if the xmlid is gone
```

Comment:
```suggestion
        return
```

This comment is not really correct in the current implementation: even without that early return it would not run the callback (thereby potentially creating the xmlid) in the `None` case (due to the `is False` check).

Reply:
The comment was just to highlight about what's (potentially) not done. Now I didn't want to rely in `force_noupdate` being aligned with this implementation (i.e. it doesn't create the xmlid, which is the case right now). Anyway I will remove the comment.